sound/soc/sof/intel/hda-pcm.c

```diff
@@ -153,7 +153,7 @@ snd_pcm_uframes_t hda_dsp_pcm_pointer(struct snd_sof_dev *sdev,
 		return 0;
 	}
 
-	if (hda && !hda->no_ipc_position) {
+	if (hda && !hda->ipc_nowakeup) {
```

Comment:
I have a conceptual issue now with having the hda structure carry information about the nowakeup when it should really be a stream operation. the rules should be:
a) the driver does not support nowakeup on any streams
b) the driver supports nowakeup on the streams requested by the application.
This structure based on hda->ipc_nowakeup makes no sense to me.

Reply:
i replied right after your comment, But it showed pending and gone.
I'm reviewing this feature again. will come up with the points you brought.